test/avocado/selenium-machines-basic.py

```diff
@@ -0,0 +1,110 @@
+import os
+from avocado import skipIf
+from testlib_avocado.seleniumlib import clickable, invisible
+from machineslib import MachinesLib
+
+
+class MachinesBasicTestSuite(MachinesLib):
+    """
+    :avocado: enable
+    :avocado: tags=machines
+    """
+
+    def testNoVm(self):
+        self.wait_text("No VM is running or defined on this host")
+
+    def testOverviewInfo(self):
+        name = "staticvm"
+        self.create_vm(name)
+
+        self.wait_in_text('256 MiB', '#vm-{}-memory'.format(name))
+        self.wait_in_text('1', '#vm-{}-vcpus-count'.format(name))
+        self.wait_in_text('custom', '#vm-{}-cputype'.format(name))
+        self.wait_in_text('pc', '#vm-{}-emulatedmachine'.format(name))
+        self.wait_in_text('disk,network', '#vm-{}-bootorder'.format(name))
+
+    def testRunVm(self):
+        name = "staticvm"
+        args = self.create_vm(name, state='shut off')
+
+        self.click(self.wait_id('vm-{}-run'.format(name), cond=clickable))
+        self.wait_in_text('running', '#vm-{}-state'.format(name))
+        self.wait_id('vm-{}-run'.format(name), cond=invisible)
+        self.wait_id('vm-{}-reboot'.format(name))
+        self.wait_id('vm-{}-off'.format(name))
+        self.wait_id('vm-{}-delete'.format(name))
+        self.wait_vm_complete_start(args)
+
+    def testRestartVm(self):
+        name = "staticvm"
+        args = self.create_vm(name, wait=True)
+
+        self.click(self.wait_id('vm-{}-reboot'.format(name), cond=clickable))
+        self.machine.execute(
+            'until sudo cat {} | grep "reboot: Power down"; do sleep 1; done'.format(args.get('logfile')))
+        self.wait_in_text('running', '#vm-{}-state'.format(name))
+
+    @skipIf(os.environ.get('HUB') == '10.111.112.10', "Doesn't work when using run-tests")
+    def testForceRestartVm(self):
+        name = "staticvm"
+        args = self.create_vm(name, wait=True)
+
+        self.click(self.wait_id('vm-{}-reboot-caret'.format(name), cond=clickable))
+        self.click(self.wait_id('vm-{}-forceReboot'.format(name), cond=clickable))
+        self.machine.execute(
+            'until sudo cat {} | grep -E "login:.*Initializing cgroup"; do sleep 1; done'.format(args.get('logfile')))
+        self.wait_in_text('running', '#vm-{}-state'.format(name))
+
+    def testShutdownVm(self):
+        name = "staticvm"
+        args = self.create_vm(name, wait=True)
+
+        self.click(self.wait_id('vm-{}-off'.format(name), cond=clickable))
+        self.wait_in_text('shut off', '#vm-{}-state'.format(name))
+        self.machine.execute(
```

Comment:
Just nitpick I saw similar code like this on more places, maybe would be good candidate to some method from it. it waits until some text appear in log.

Reply:
The test won't hang, because self.machine.execute() has a timeout parameter, it defaults to 120s, if the cmd didn't return within 120s, self.machine.execute() will raise an error.
I see there is a function wait(func, msg=None, delay=1, tries=60) in testlib_avocado.common.testlib, which could fulfil the requirement, I tried to import it, but failed with no tap package. So I wonder if it's appropriate to import from that file, if not, how about copy the wait() function to testlib_avocado.timeoutlib